osu.Game.Rulesets.Catch/UI/PoolableHitExplosion.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Pooling;
+using osu.Game.Rulesets.Catch.Objects;
+using osu.Game.Rulesets.Judgements;
+using osu.Game.Skinning;
+using osuTK.Graphics;
+
+namespace osu.Game.Rulesets.Catch.UI
+{
+    public class PoolableHitExplosion : PoolableDrawable
+    {
+        public const double DURATION = 1000;
+
+        private Color4 objectColour;
+
+        public Color4 ObjectColour
+        {
+            get => objectColour;
+            set
+            {
+                if (objectColour == value)
+                    return;
+
+                objectColour = value;
+
+                if (skinnableExplosion.Drawable is ICatchHitExplosion hitExplosion)
+                {
+                    hitExplosion.ObjectColour = value;
+                }
+            }
+        }
+
+        public PalpableCatchHitObject HitObject;
+
+        public JudgementResult JudgementResult;
+
+        public float CatcherWidth;
+
+        public float CatchPosition;
+
+        private SkinnableDrawable skinnableExplosion;
+
+        public PoolableHitExplosion()
+        {
+            RelativeSizeAxes = Axes.Both;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            InternalChild = skinnableExplosion = new SkinnableDrawable(new CatchSkinComponent(CatchSkinComponents.HitExplosion), _ => new DefaultHitExplosion());
+        }
+
+        protected override void PrepareForUse()
+        {
+            base.PrepareForUse();
+
+            if (skinnableExplosion?.Drawable is CatchHitExplosion explosion)
+            {
+                explosion.JudgementResult = JudgementResult;
+                explosion.ObjectColour = ObjectColour;
+                explosion.HitObject = HitObject;
+                explosion.CatchPosition = CatchPosition;
+                explosion.CatcherWidth = CatcherWidth;
```

Comment:
I'd like to get rid of this second layer of data pushdown, if possible. I think the `PoolableHitExplosion` should cache itself so that its children can retrieve it via DI, read the values themselves, and adjust appropriately (probably at the start of `RunAnimation()`?)

Reply:
I've added this in 16638a28150c122ac842b3deae1993756c2d4806